fs/operations/operations.go

```diff
@@ -361,10 +361,21 @@ func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Obj
 	if !ci.Inplace && f.Features().Move != nil && f.Features().PartialUploads && !strings.HasSuffix(remote, ".rclonelink") {
 		// Avoid making the leaf name longer if it's already lengthy to avoid
 		// trouble with file name length limits.
-		suffix := "." + random.String(8) + ".partial"
+		ext := strings.TrimSpace(ci.PartialUploadExtension)
+		if len(ext) == 0 {
+			ext = fs.GetConfig(context.Background()).PartialUploadExtension
+		}
+
+		suffix := "." + random.String(8) + "." + ext
 		base := path.Base(remotePartial)
 		if len(base) > 100 {
-			remotePartial = remotePartial[:len(remotePartial)-len(suffix)] + suffix
+			limit := len(base) - len(suffix)
+
+			if len(suffix) > len(base) {
```

Comment:
This can never be true since len(base) > 100 and we know len(suffix) = len("." + random.String(8) + "." + ext) <= 25

Reply:
I've added all that checks for those cases, when `rclone` is used as library, and not as `cmd`. So the user won't be able to break execution even if they really want 🙂
I am still worried about that case. Should the user be responsible for value they use?

Reply:
If you want to check the length is <= 15 in the function itself then that is fine - do that and return an error.